plugins/catalog-backend/src/ingestion/types.ts

```diff
@@ -98,3 +99,17 @@ export type AnalyzeLocationEntityField = {
    */
   description: string;
 };
+
+/** @public */
+export type AnalyzeOptions = {
+  url: string;
+  catalogFilename?: string;
+};
+
+/** @public */
+export type ScmLocationAnalyzer = {
+  /** The integration type this location analyzer can work with */
+  getIntegrationType(): string;
```

Comment:
Some other places in the code where things are flexible over a range of possibilities, invert the control and instead have a predicate. So this method could be replaced with `supports(url: string): boolean` or similar instead. Then that code could choose to check with ScmIntegrations whether the type for that url is something that it supports, or any other possible heuristic.